src/libraries/System.Private.CoreLib/src/System/Resources/ResourceManager.cs

```diff
@@ -227,9 +226,13 @@ public ResourceManager(Type resourceSource)
             if (resourceSource is not RuntimeType)
                 throw new ArgumentException(SR.Argument_MustBeRuntimeType);
 
-            _locationInfo = resourceSource;
-            MainAssembly = _locationInfo.Assembly;
-            BaseNameField = resourceSource.Name;
+            MainAssembly = resourceSource.Assembly;
+
+            string? nameSpace = resourceSource.Namespace;
+            char c = Type.Delimiter;
+            BaseNameField = nameSpace is null
+                ? resourceSource.Name
+                : string.Concat(nameSpace, new ReadOnlySpan<char>(in c), resourceSource.Name);
```

Comment:
This is the key change. Previously, we'd store the type and set `BaseNameField` to be the type name. Then when looking up resources we'd pass in both the type and the name and `GetManifestResourceStream` would recombine them. Now, we just compute the correct base name upfront, which means we don't need to store the type at all.

One downside to this approach is that the change is observable to someone who is deriving from `ResourceManager` and modifying the `BaseNameField` field which is `protected`. However, I can't imagine a realistic use-case for that; having `BaseName` return the right/consistent thing in all cases feels more valuable.

Reply:
Why could not use just the `FullName`? i.e. `BaseNameField = resourceSource.FullName!`

Reply:
Quite possibly we could do that and I'm happy to make the change. The reason I didn't do that initially is that the old code used this logic rather than `FullName` ; I thought there might be a reason for that.

Reply:
Though in that case the resource name populated for generic types will be different:
- `FullName`: System.Collections.Generic.List`1[[System.String, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]
- `Namespace + '.' + Name`: System.Collections.Generic.List`1

Not really sure we should cover this scenario

Reply:
What was the old behavior when using the generic types? I prefer to reduce any breaking change here.

Reply:
By old behavior the resource name will be "System.Collections.Generic.List`1"